src/error.rs

```diff
@@ -0,0 +1,66 @@
+use std::error::Error as TError;
+use std::io;
+use std::fmt;
+use std::path::PathBuf;
+
+#[derive(Debug)]
+pub enum Error {
+  Load(io::Error),
+  Copy(io::Error, PathBuf, PathBuf),
+  CopyTimeOut(PathBuf, PathBuf),
+  Find(String)
+}
+
+// TODO: Figure out a way to compare the insides so that the Eq is correct.
```

Comment:
Would it really be necessary for the equality here to be 100% correct? As far as I saw it only needs it so that it knows which error is which.